### ISSUE TITLE
MUI accessibility fixes

### ISSUE THREAD

**Issue by mayank99** [iTwin/stratakit#1185]
Tracking all accessibility issues that are fixable in the MUI theme.

**Comment by GerardasB** [comment]
Issues noticed during **Weekly accessibility review 2026-01-27**:

- [x] **Autocomplete**: "clear" button is not keyboard accessible/focusable https://github.com/iTwin/stratakit/pull/1213
- [x] **Autocomplete**: "clear" button should always be visible https://github.com/iTwin/stratakit/pull/1214
- [x] **Autocomplete**: multi-select lacks grouping (consider fieldset or group pattern) https://github.com/iTwin/stratakit/pull/1215
- [x] **Autocomplete**: multi-select options should be focusable https://github.com/iTwin/stratakit/pull/1252
- [x] **Autocomplete**: "chip" itself should not be clickable/focusable. Instead "chip dismiss button" shuld be focusable (chip should contain a "Delete" button) https://github.com/iTwin/stratakit/pull/1252
- [ ] **Autocomplete**: multi-select auto-close is inconsistent with `Select`
- [ ] **Select**: verify if it's correct for focus to jump to last selected option
- [ ] **Select**: option focus state (should announce selected options, instead of "not selected")

Additional improvements after further internal discussions:
- [ ] **Autocomplete**: chip buttons should be rendered outside of the input (after the input).
